### ISSUE TITLE
Let sparse_softmax_cross_entropy_with_logits accept int32 labels?

### ISSUE THREAD

**Issue by rdipietro** [tensorflow/tensorflow#1250]
Right now `sparse_softmax_cross_entropy_with_logits` only accepts `labels` as an int64 `Tensor`. Is this intentional? Why would we want to avoid int32 `labels`?

**Comment by ebrevdo** [comment]
This is due to a deep technical issue with being able to access the values on a GPU.  For now, use tf.to_int64 to convert your values.

**Comment by ebrevdo** [comment]
Closed because the deeper issue has an open internal bug case.

**Comment by girving** [comment]
int32 should work fine on GPUs now.  I'll fix this.
